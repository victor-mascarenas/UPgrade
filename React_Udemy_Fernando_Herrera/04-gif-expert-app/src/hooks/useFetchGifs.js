import { useState } from "react";

const useFetchGifs = () => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });
    return state;
};

export default useFetchGifs;