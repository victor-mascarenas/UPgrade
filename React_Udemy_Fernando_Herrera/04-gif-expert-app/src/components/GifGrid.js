import React/* , {useState, useEffect} */ from 'react';
import useFetchGifs from '../hooks/useFetchGifs';
//import getGifs from '../helpers/getGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({category}) => {
    /* const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs(category)
            .then(setImages);
    }, [category]); */

    const {data: images, loading} = useFetchGifs(category);

    return(
        <>
            <h3 className='animate_animated animate_fadeIn'>{category}</h3>
            {loading && <p className='animate_animated animate_flash'>Loading...</p>}
            <div className="card-grid">
                {
                    images.map(img => {
                        return <GifGridItem key={img.id} {...img}/>;
                    })
                }
            </div>
        </>
    );
};
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;