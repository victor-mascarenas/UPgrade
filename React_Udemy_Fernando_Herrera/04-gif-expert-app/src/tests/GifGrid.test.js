import React from "react";
import {shallow} from 'enzyme';
import GifGrid from '../components/GifGrid';
import useFetchGifs from "../hooks/useFetchGifs";

//'Falsea' el hook
jest.mock("../hooks/useFetchGifs");

describe('Pruebas en el componente GifGrid', () => {
    const category = 'Naruto';
    const mockReturn = {
        data: [],
        loading: true
    };
    let wrapper;
    
    beforeEach(() => {
        //Se establece el valor de retorno del hook false
        useFetchGifs.mockReturnValue(mockReturn);
        wrapper = shallow(<GifGrid category={category}/>);
    });

    test('Debe empatar con el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        const mockReturn = {
            data: [{
                id: 'ABC',
                url: 'https://sdsdf/sdf/etr/img.gif',
                title: 'Imagen'
            }],
            loading: false
        };
        useFetchGifs.mockReturnValue(mockReturn);
        wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
});
