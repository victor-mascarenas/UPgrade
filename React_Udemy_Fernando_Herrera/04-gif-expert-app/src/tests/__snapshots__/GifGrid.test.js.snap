// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate_animated animate_fadeIn"
  >
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Imagen"
      url="https://sdsdf/sdf/etr/img.gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe empatar con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate_animated animate_fadeIn"
  >
    Naruto
  </h3>
  <p
    className="animate_animated animate_flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
